### ISSUE TITLE
Factor of 0.5 in reported MSE metric

### ISSUE THREAD

**Issue by rishabh-ranjan** [cszhangzhen/H2MN#2]
Why is there a factor of 0.5 in [https://github.com/cszhangzhen/H2MN/blob/a2cf278bb5d4f8b4bf2c01f8c3e67f1150b0b940/main_regression.py#L156](url)?

The MSE metric described in the paper does not document this, which is misleading. Please clarify.

**Comment by cszhangzhen** [comment]
Hi, we follow the metric in SimGNN paper. Please refer to [https://github.com/yunshengb/SimGNN/blob/master/src/metrics.py#L69](url)

Sorry for not describing it clearly in the paper.

**Comment by AAAAzhu** [comment]
Hello author, after reading your answer I still have some doubts, why this paper needs to multiply 0.5 when calculating mse, obviously the result of F.mse_loss is the value of mse after averaging.

**Comment by AAAAzhu** [comment]
> Hi, we follow the metric in SimGNN paper. Please refer to [https://github.com/yunshengb/SimGNN/blob/master/src/metrics.py#L69](url)
> 
> Sorry for not describing it clearly in the paper.

Hello author, after reading your answer I still have some doubts, why this paper needs to multiply 0.5 when calculating mse, obviously the result of F.mse_loss is the value of mse after averaging.

**Comment by cszhangzhen** [comment]
There are no specific reasons. We simply follow the metric used in SimGNN paper.

**Comment by AAAAzhu** [comment]
> https://github.com/yunshengb/SimGNN/blob/master/src/metrics.py#L69

Thank you for your response, but I'm not sure about the rationale behind this approach. Additionally, during the comparison with baselines, other models did not multiply by 0.5 at the end. Could there be an appropriate explanation for this?

**Comment by cszhangzhen** [comment]
I think it is fine to multiply 0.5 or not , as long as all the baselines use the same metric.

Multiplying 0.5 in front of MSE is for mathematical convenience in traditional machine learning. When training a model using gradient-based optimization, the loss function is given by:
$MSE = \frac{1}{n}\sum_{i=1}^n(y_i-\hat{y}_i)$

Without the 0.5 factor, the derivative of this loss with respect to a model parameter $w$ would be:
$\frac{d}{dw} \left( \frac{1}{n} \sum_{i=1}^{n} (y_i - \hat{y}_i)^2 \right) = 2(y_i - \hat{y}_i)$

With the 0.5 factor, the loss becomes:
$\frac{1}{n} \sum_{i=1}^{n} 0.5 (y_i - \hat{y}_i)^2$

The derivative with respect to $w$ is simplified to:
$\frac{d}{dw} \left( \frac{1}{n} \sum_{i=1}^{n} 0.5 (y_i - \hat{y}_i)^2 \right) = (y_i - \hat{y}_i)$

I think this is a established practice in traditional machine learning, thus the SimGNN authors use it in the evaluation.

**Comment by AAAAzhu** [comment]
> I think it is fine to multiply 0.5 or not , as long as all the baselines use the same metric.
> 
> Multiplying 0.5 in front of MSE is for mathematical convenience in traditional machine learning. When training a model using gradient-based optimization, the loss function is given by: M S E = 1 n ∑ i = 1 n ( y i − y i )
> 
> Without the 0.5 factor, the derivative of this loss with respect to a model parameter w would be: d d w ( 1 n ∑ i = 1 n ( y i − y i ) 2 ) = 2 ( y i − y i )
> 
> With the 0.5 factor, the loss becomes: 1 n ∑ i = 1 n 0.5 ( y i − y i ) 2
> 
> The derivative with respect to w is simplified to: d d w ( 1 n ∑ i = 1 n 0.5 ( y i − y i ) 2 ) = ( y i − y i )
> 
> I think this is a established practice in traditional machine learning, thus the SimGNN authors use it in the evaluation.

Thank you for your prompt response, which has helped alleviate some of my confusion. Are you suggesting that all the GNN baselines used in your comparative experiments, like SimGNN, also multiply the MSE result by 0.5? Furthermore, since more recent studies no longer apply this multiplication when calculating results, should the MSE in this paper be doubled during comparisons to ensure consistency?

**Comment by cszhangzhen** [comment]
Both the approaches are acceptable, as long as the evaluation metric is consistent across all the baselines.

**Comment by AAAAzhu** [comment]
> Both the approaches are acceptable, as long as the evaluation metric is consistent across all the baselines.

你好，那实验中的基线是你重新跑的吗，我想确定一下基线方法是否都乘以了0.5

**Comment by cszhangzhen** [comment]
是的，我用的和SimGNN的setting一样的
